tests/mod_test.py

```diff
@@ -0,0 +1,75 @@
+from .context import assert_equal
+import pytest
+from sympy import Symbol, Rational, Mod, sqrt, nsimplify, pi, GoldenRatio
+from sympy.physics.units import hbar
+
+x = Symbol('x', real=True)
+y = Symbol('y', real=True)
+
+
+def test_mod_usual():
```

Comment:
@tirkarp this is an awesome test suite! nice job!

Reply:
We might consider adding a test to the all_bad_test.py to check that we get the matrix operand error.

Reply:
Added.